flutter-gui-tests/testSrc/io/flutter/tests/gui/fixtures/FlutterInspectorFixture.kt

```diff
@@ -54,16 +91,92 @@ class FlutterInspectorFixture(project: Project, robot: Robot, private val ideFra
 
     val notificationPanel = inspectorPanelRef.get()
     assertNotNull(notificationPanel)
-    return InspectorPanelFixture(notificationPanel)
+    return InspectorPanelFixture(notificationPanel, type)
   }
 
   private fun findInspectorPanel(type: InspectorService.FlutterTreeType): InspectorPanel? {
-    val panels = ideFrame.robot().finder().findAll(contents[0].component,
-                                                   ClassNameMatcher.forClass("io.flutter.view.InspectorPanel", JPanel::class.java))
-    return panels.firstOrNull { (it as InspectorPanel).treeType == type } as InspectorPanel
+    val panels = finder().findAll(contents[0].component, classMatcher("io.flutter.view.InspectorPanel", JPanel::class.java))
+    return panels.firstOrNull { it is InspectorPanel && it.treeType == type && !it.isDetailsSubtree } as InspectorPanel
   }
 
-  inner class InspectorPanelFixture(val inspectorPanel: InspectorPanel) {
+  // The InspectorPanel is a little tricky to work with. In order for the tree to have content its view must be made
+  // visible (either Widgets or Render Tree). However, we don't want to return a reference to an empty tree, because
+  // of timing issues. We use the fact that the tree has content as a signal to move on to the next step.
+  inner class InspectorPanelFixture(val inspectorPanel: InspectorPanel, val type: InspectorService.FlutterTreeType) {
+
+    fun show() {
+      showTab(tabIndex())
+    }
+
+    private fun tabIndex(): Int {
+      return if (type == InspectorService.FlutterTreeType.widget) 0 else 1
+    }
+
+    fun inspectorTreeFixture(isDetails: Boolean = false): InspectorTreeFixture {
+      val inspectorTreeRef = Ref<InspectorTree>()
 
+      pause(object : Condition("Tree shows up") {
+        override fun test(): Boolean {
+          val inspectorTree = findInspectorTree(isDetails)
+          inspectorTreeRef.set(inspectorTree)
+          return inspectorTree != null
+        }
+      }, Timeouts.seconds10)
+
+      val inspectorTree = inspectorTreeRef.get()
+      assertNotNull(inspectorTree)
+      return InspectorTreeFixture(inspectorTree)
+    }
+
+    fun findInspectorTree(isDetails: Boolean): InspectorTree? {
+      val trees = finder().findAll(inspectorPanel, classMatcher("io.flutter.inspector.InspectorTree", JTree::class.java))
+      val tree = trees.firstOrNull { it is InspectorTree && isDetails == it.detailsSubtree }
+      if (tree != null) return tree as InspectorTree else return null
+    }
+  }
+
+  // This fixture is used to access the tree for both Widgets and Render Objects.
+  inner class InspectorTreeFixture(private val inspectorTree: InspectorTree) {
+
+    fun treeFixture(): JTreeFixture {
+      return JTreeFixture(ideFrame.robot(), inspectorTree)
+    }
+
+    fun selectRow(number: Int, expand: Boolean = true) {
+      waitForContent()
+      treeFixture().clickRow(number) // This should not collapse the tree, but it does.
```

Comment:
It looks like an off-by-one error. The click occurs exactly on the left-side boundary of the row bounding box. That rectangle apparently overlaps the bounding box of the triangle icon that expands and collapses the sub-tree. I don't think it is a framework error because I can, with sufficient care, reproduce the problem manually. OTOH if the click was in the center of the bounding box there'd be no problem, but that's a choice made by fest, a library used by the testing framework.